09_serialization.md

```diff
@@ -30,8 +30,64 @@ There is a subtle difference in `pickle` imports between Python 2 and Python 3.
 
 ### PySpark Serialization Strategies
 
+#### AutoBatchedSerializer
+
+`AutoBatchedSerializer` is used to serialize a stream of objects in batches of size which is dynamically adjusted on runtime to keep the size of an individual batch in bytes close to the `bestSize` parameter. By default it is 65536 bytes. The general algorithm can be described as follows:
+
+1. Set `batchSize` to 1
+2. While input stream is not empty
+
+  - read `batchSize` elements and serialize,
+  - compute `size` in bytes of the serialized batch,
+    - if `size` is less than `bestSize` set `batchSize` to `batchSize` * 2
+    - else if `size` is greater than `bestSize` set `batchSize` to `batchSize` / 2
+    - else keep current value
+
+
+#### BatchedSerializer
+
+`BatchedSerializer` is used to serialize a stream of objects using either fixed size batches or unlimited batches.
+
 ### Configuring PySpark Serialization
 
+By default PySpark uses `AutoBatchedSerializer` with `PickleSerializer`. Global serialization mechanism can be configured during `SparkContext` initialization and provides two configurable properties - `batchSize` and `serializer`.
+
+
+- `batchsize` argument is used to configure batching strategy and takes an intger which has following interpretation:
+
+  - `0` -  `AutoBatchedSerializer`.
+  - `-1` - `BatchedSerializer` with unlimited batch size.
+  - positive integer - `BatchedSerializer` with fixed batch size.
+
+
+- `serializer` accepts an instance of a `pyspark.serializer` for example:
+
+
+  ```python
+  from pyspark import SparkContext
+  from pyspark.serializers import MarshalSerializer
+
+  sc = SparkContext(serializer=MarshalSerializer())
+  ```
+
+In Spark 2.0.0+, you have to create `SparkContext` before `SparkSession` if you builder or pass it explicitly to `SparkSession` constructor.
```

Comment:
Me neither ;) To be serious it should be something like this: "_In Spark 2.0.0+ you have to create `SparkContext` before `SparkSession` if you use `SparkSession` builder or pass it explicitly to `SparkSession` constructor otherwise._"